tools/Dashboard/DevHome.Dashboard/Views/DashboardView.xaml.cs

```diff
@@ -358,6 +382,28 @@ private async void RestorePinnedWidgets(object sender, RoutedEventArgs e)
                     Log.Logger()?.ReportError("DashboardView", $"RestorePinnedWidgets(): ", ex);
                 }
             }
+
+            // Now that we've ordered the widgets, put them in their final collection.
+            var finalPlace = 0;
+            foreach (var orderedWidget in restoredWidgetsWithPosition)
+            {
+                var widget = orderedWidget.Value;
+                var size = await widget.GetSizeAsync();
+                await InsertWidgetInPinnedWidgetsAsync(widget, size, finalPlace);

```

Comment:
`PinnedWidgets` is static, hence concurrent execution of async `OnLoad` method (e.g. switch pages quickly) could potentially insert the same set of widgets to the page multiple times.

Reply:
This is a great point. Since it's already an issue and I want to give it some thought, I opened #1215